includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -2,20 +2,44 @@
 
 class AMP_Image_Dimension_Extractor {
 	static $callbacks_registered = false;
+    const STATUS_FAILED_LAST_ATTEMPT = 'failed';
+    const STATUS_IMAGE_EXTRACTION_FAILED= 'failed';
 
-	static public function extract( $url ) {
+    static public function extract( $urls ) {
 		if ( ! self::$callbacks_registered ) {
 			self::register_callbacks();
 		}
 
-		$url = self::normalize_url( $url );
-		if ( false === $url ) {
-			return false;
-		}
+		foreach ($urls as &$url) {
+            $url = self::normalize_url($url);
+        }
 
-		return apply_filters( 'amp_extract_image_dimensions', false, $url );
+		$dimensions = apply_filters( 'amp_extract_image_dimensions', $urls );
+
+        if ( has_filter('amp_extract_image_dimensions') ) {
+            return $dimensions;
+        } else {
+            return self::modify_return_value_for_tests_that_disable_extraction_filter($dimensions);
```

Comment:
The functions normally invoked by the amp_extract_image_dimensions filter modify the format/content of the array returned by AMP_Image_Dimension_Extractor::extract. When the filter is disabled, extract() returns an array that code downstream from it isn't expecting/can't properly parse. modify_return_value_for_tests_that_disable_extraction_filter() normalized the array in such a scenario so that it would work properly downstream. However, revisiting this code now, though the function name is descriptive, it's possible the filter may be removed/disabled for some other reason, so I will give the function a more general name.